### PR TITLE
Apply xformers in diffusers_trainer.py

### PR DESCRIPTION
**Pull request by laksjdjf**
apply xformers to reduce VRAM usage and improve performance.

GPU=A6000×1, batch size=2, resolution=768, step=1310
W B Chart 2022_11_9 10_23_51

### CODE REVIEW

**Comment by harubaru** [comment]
Are there any extra dependencies required for this change?

File: diffusers_trainer.py [hunk 1]
```diff
@@ -81,6 +81,7 @@
 parser.add_argument('--image_log_scheduler', type=str, default="PNDMScheduler", help='Number of inference steps to use to log images.')
 parser.add_argument('--clip_penultimate', type=bool, default=False, help='Use penultimate CLIP layer for text embedding')
 parser.add_argument('--output_bucket_info', type=bool, default=False, help='Outputs bucket information and exits')
+parser.add_argument('--use_xformers', action='store_true',help='Use memory efficient attention')
```
Comment (harubaru) [inline diffusers_trainer.py:84]: Change the argument type from a `store_true` to a `bool` for consistency

**Comment by laksjdjf** [comment]
> Are there any extra dependencies required for this change?

it needs to install xformers library.
https://github.com/facebookresearch/xformers

I did below in Paperspace gradient. it takes an hour.
```
git clone https://github.com/facebookresearch/xformers.git
cd xformers
git submodule update --init --recursive
pip install -r requirements.txt
python setup.py bdist_wheel
pip install dist/xformers-0.0.14.dev0-cp39-cp39-linux_x86_64.whl
```

**Comment by laksjdjf** [comment]
memory_efficient_attention is a new feature in diffusers 0.7
https://github.com/huggingface/diffusers/releases/tag/v0.7.0

**Pull request merged**